rootex/core/language-locale.cpp

```diff
@@ -0,0 +1,31 @@
+#include "core/language-locale.h"
+
+Locale* Locale::GetSingleton()
+{
+	static Locale singleton;
+	return &singleton;
+}
+
+void Locale::LoadLanguage(String location)
+{
+	if (OS::IsFile(location))
+	{
+		JSON::json file;
+		std::ifstream i(location);
+		i >> file;
+		m_GameStrings = file.get<std::unordered_map<String, String>>();
+	}
+}
+
+String Locale::GetString(String key)
+{
+	return m_GameStrings.at(key);
+}
+
+Locale::Locale()
```

Comment:
Empty constructors and destructors should by default by `Constructor() = default` in the header file. No need to declare them here if you are not going to do anything in them